src/Interpreters/MonotonicityCheckVisitor.h

```diff
@@ -70,6 +70,9 @@ class MonotonicityCheckMatcher
             if (!pos)
                 return false;
 
+            if (*pos >= tables.size())
```

Comment:
I don't understand the code.
Why one of the previous functions can return invalid "pos"?

Reply:
Because `IdentifierSemantic::getMembership` returns a position stored in AST. This position happened to be invalid while we were trying to re-analyze a part of AST somewhere from StorageMerge.

Reply:
Basically I hope that we remove IdentifierSemantic after Analyzer is merged. It is vary hard to use it reliably.